data_rentgen/db/repositories/output.py

```diff
@@ -83,6 +162,66 @@ async def list_by_dataset_ids(
         result = await self._session.scalars(query)
         return list(result.all())
 
+    async def list_by_dataset_ids_grouped_by_run(
```

Comment:
What if instead of adding 3 copies of the same method, we'll add an optional `group_by` method? Examples:
```python
dataset.list_by_dataset_ids([...], group_by=Operation)
dataset.list_by_dataset_ids([...], group_by=Run)
# or
dataset.list_by_dataset_ids([...], group_by=[Dataset.operation_id, Dataset.run_id])
dataset.list_by_dataset_ids([...], group_by=[Dataset.run_id])
```

Reply:
I am not sure about unpacking result for this methods.
One row is tuple and we need to unpack it to create Input/Output instance. But for different params we would return different nums of elements in row:
For example:

```python
output.list_by_dataset_ids([...], group_by=Run) ->
Output(
                created_at=None,
                id=None,
                operation_id=None,
                job_id=None,
                run_id=row[0],
                dataset_id=row[1],
                num_bytes=row[2],
                num_rows=row[3],
                num_files=row[4],
            )

output.list_by_dataset_ids([...], group_by=Job) ->

Output(
                created_at=None,
                id=None,
                operation_id=None,
                job_id=row[0],
                run_id=None,
                dataset_id=row[1],
                num_bytes=row[2],
                num_rows=row[3],
                num_files=row[4],
            )

```

Reply:
```python
output.list_by_dataset_ids([...], group_by=Run):
output.list_by_dataset_ids([...], group_by=Job):

query_group_by = [Dataset.id]
if group_by == Operation:
  query_group_by.append(Dataset.operation_id)
if group_by == Run:
  query_group_by.append(Dataset.run_id)
if group_by == Job:
  query_group_by.append(Dataset.job_id)

query = (
  select(
    Output.id if group_by == Operation else literal(None).alias("id"),
    Output.operation_id if group_by == Operation else literal(None).alias("operation_id"),
    Output.run_id if group_by in (Operation, Run) else literal(None).alias("run_id"),
    Output.job_id,
    Output.dataset_id,
    sum(Output.num_bytes).alias("num_bytes"),
    sum(Output.num_files).alias("num_files"),
    sum(Output.num_rows).alias("num_rows"),
  )
  .where(...)
  .group_by(*query_group_by)
```

And then convert Row to Output